sdk/dotnet/Pulumi.Automation/Commands/LocalPulumiCmd.cs

```diff
@@ -25,7 +25,12 @@ internal class LocalPulumiCmd : IPulumiCmd
             // this causes commands to fail rather than prompting for input (and thus hanging indefinitely)
             var completeArgs = args.Concat(new[] { "--non-interactive" });
 
-            var env = new Dictionary<string, string>();
+            var env = new Dictionary<string, string>()
+            {
+                // Required for event log
+                ["PULUMI_DEBUG_COMMANDS"] = "true",
```

Comment:
I moved setting of the variable to `Workspace` and behind a conditional

Reply:
That works. But I think the conditional can just check if the delegate is null rather than iterating over the Args. And I think we should find a way to not always instantiate the log watcher even when we don't need it. It seems weird to instantiate a type that may or may not do anything

Reply:
That may be simpler if you make it lower level. For instance if you pass the nullable OnEvent action all the way down to LocalPulumiCmd and have that be the single place that your event log watcher is conditionally instantiated